### ISSUE TITLE
Rule proposal: `no-useless-nesting-in-template-literals`

### ISSUE THREAD

**Issue by B2o5T** [sindresorhus/eslint-plugin-unicorn#1652]
### Description

Reports nested instances of a string or a template literal. Suggests inlining the nested instances into the containing template string.
inspired by JetBrains https://www.jetbrains.com/help/phpstorm/javascript-and-typescript-redundant-nesting-in-template-literal.html

### Fail

```js
`Hello, ${`Brave ${"New"}`} ${"World"}!`

`${2021} year!`

`${true} or ${false}`

`${String(someVar)} foo` // redundant call String as .toString() will be applied automatically
```

### Pass

```js
`Hello, Brave New World!`

`2021 year!`

`true or false`

`${someVar} foo`
```

**Comment by sindresorhus** [comment]
For consistency, it should be called `no-useless-nesting-in-template-literals` (also note the `s` at the end)

**Comment by fisker** [comment]
Is this duplicate of #1261?
